src/components/AsyncButton.tsx

```diff
@@ -19,7 +19,9 @@ import React, { useCallback, useEffect, useRef, useState } from "react";
 import { Button, type ButtonProps } from "react-bootstrap";
 
 export type AsyncButtonProps = ButtonProps & {
-  onClick: (() => Promise<void>) | (() => void);
+  onClick:
+    | ((event: React.MouseEvent) => Promise<void>)
+    | ((event: React.MouseEvent) => void);
```

Comment:
Can this be simplified?
```suggestion
  onClick: (event: React.MouseEvent) => void | Promise<void>
```

Reply:
Added a comment -- it makes breaks the call-site here: https://github.com/pixiebrix/pixiebrix-extension/blob/0376d9f175f4bc7033b33a60e1bca8b2e235ab65/src/contrib/google/sheets/RequireGoogleApi.tsx#L60-L60